### PR TITLE
init Tekton backend commit

### PR DESCRIPTION
**Pull request by Tomcli**
Initial commit for the Tekton KFP server and persistent agent. We will add the instructions in the follow up PR since we want to have version control on this initial code commit.

- Modified api docker build to use go modules
- Moved all the Argo specific API to Tekton specific API (Still need to implement it with polymorphism or abstracted function as the ideal solution)
   - workflow type changed to pipelineRun type
   - Argo v1alpha1 api changed to Tekton v1beta1 api
   - param value type changed from Argo string to Tekton ArrayOrString Type
- go.mod is updated to match with the Tekton dependencies.
- go.sum is updated for dependency version control.
- Commented out Argo specific tests to skip type check errors.

### CODE REVIEW

**Comment by Tomcli** [comment]
The Travis test is for the python SDK and it's funky since it sometimes passes the tests.
/retest

File: backend/Dockerfile [hunk 1]
```diff
@@ -60,7 +66,7 @@ RUN line="import kfp;kfp.components.default_base_image_or_builder='gcr.io/google
     python3 "$pipeline"; \
     done
 
-FROM debian:stretch
+FROM golang:1.13.0
```
Comment (animeshsingh) [inline backend/Dockerfile:69]: Can we document why we need this?

File: backend/src/agent/persistence/client/workflow_client.go [hunk 2]
```diff
@@ -27,11 +27,11 @@ type WorkflowClientInterface interface {
 
 // WorkflowClient is a client to call the Workflow API.
 type WorkflowClient struct {
-	informer v1alpha1.WorkflowInformer
+	informer v1beta1.PipelineRunInformer
```
Comment (animeshsingh) [inline backend/src/agent/persistence/client/workflow_client.go:30]: Again a comment mentioning the switch, and we will come back and define a common object

File: backend/src/common/util/workflow.go [hunk 3]
```diff
@@ -143,33 +146,37 @@ func (w *Workflow) ScheduledAtInSecOr0() int64 {
 }
 
 func (w *Workflow) FinishedAt() int64 {
-	if w.Status.FinishedAt.IsZero() {
+	if w.Status.PipelineRunStatusFields.CompletionTime.IsZero() {
```
Comment (animeshsingh) [inline backend/src/common/util/workflow.go:149]: is FinishedAt.IsZero() specific to Argo?

**Pull request merged**